RA-26-Add-ViewModel-to-Other-Fragments

DenisGubernatorov opened the pull request:
Рефакторинг  RecipesListFragment, FavoritesFragment, CategoriesListFragment в соответствии с MVVM

DenisGubernatorov committed:
RA-26-Add-ViewModel-to-Other-Fragments

1) IngredientsAdapter  и MethodAdapter добавлены методы для обновления dataSet с оповещением об изменении.
2) В RecipeFragment в обсервере оповещение адпатеров происходит через новые

DenisGubernatorov committed:
RA-26-Add-ViewModel-to-Other-Fragments

1) Перевод на MVVM фрагмента RecipesListFragment

DenisGubernatorov committed:
RA-26-Add-ViewModel-to-Other-Fragments

1) Перенос recipeListAdapter в общий пакет для ui. Так как он используется в двух фрагментах

DenisGubernatorov committed:
RA-26-Add-ViewModel-to-Other-Fragments

1) Реформат FavoritesFragment в соответствии с MVVM.
2) Добавлен класс FavoritesViewModel отвечающий за обработку стейта

DenisGubernatorov committed:
RA-26-Add-ViewModel-to-Other-Fragments

1) Рефакторинг CategoriesListFragment в соответствии с MVVM.
2) Добавлен класс FavoritesViewModel отвечающий за обработку стейта

aGroupoid commented on app/src/main/java/com/example/recipesapp/ui/recipes/favorites/FavoritesViewModel.kt:
```diff
@@ -0,0 +1,57 @@
+        val favoritesList: List<Recipe>,
+        val favoritesImage: Drawable?,
+        val emptyListVisibility: Int,
+        val rvFavoritesVisibility: Int,
```
Все параметры кроме favoritesList - лишние. favoritesImage - всегда одна и та же картинка, это не динамические данные. emptyListVisibility и rvFavoritesVisibility- это реакция ui пустой или полный список favoritesList

aGroupoid commented on app/src/main/java/com/example/recipesapp/ui/recipes/recipeslist/RecipeListViewModel.kt:
```diff
@@ -0,0 +1,49 @@
+    }
+
+    data class RecipeListViewModelState(
+        val categoryList: List<Recipe>,
```
По сути это recipeList а не categoryList

aGroupoid reviewed (changes requested):
В целом ок, поправь эти комменты

DenisGubernatorov committed:
RA-26-Add-ViewModel-to-Other-Fragments

1) Изменена версия gradle согласно рекомендациям идешке
2) FavoritesViewModel из стейта убраны лишние поля и перенесены в сам фрагмент
3) Переименова список рецептов в стейте для RecipeListViewModel

aGroupoid reviewed (approved):
Всё верно, молодец
